src/Visitor.ts

```diff
@@ -0,0 +1,234 @@
+import * as ts from 'typescript'
+
+import { Counter } from './Counter'
+import { Descriptor } from './Descriptor'
+import { Input } from './Input'
+import * as lsif from './lsif'
+import { LsifSymbol } from './LsifSymbol'
+import { lsif_typed } from './main'
+import { Packages } from './Packages'
+import { Range } from './Range'
+
+export class Visitor {
```

Comment:
WDYT about giving this type a more descriptive name? E.g. we could rename `Indexer` to `ProjectIndexer`, and call this `FileIndexer`.